### PR TITLE
Added Multistage-graph.md

### LINKED ISSUE #1279: [Feature]: <New Algorithm addition>Multistage graph in dynamic programming
### Idea Title

Multistage graph

### Name:  
```[NEW ALGORITHM]```  MULTISTAGE GRAPH
You are given a directed multistage graph with n nodes. The graph is divided into multiple stages, and you need to find the shortest path from the source node (node 0) to the destination node (node n-1) while passing through intermediate stages. Each node belongs to a particular stage, and edges between nodes have associated costs. Use Dynamic Programming to solve this.

### About:  
One of the most important program in dynamic programming. 
The graph is represented as an adjacency matrix, where graph[i][j] denotes the cost of moving from node i to node j:
graph[i][j] = -1 indicates there is no edge from node i to node j.
graph[i][j] ≥ 0 represents the cost of traveling from node i to node j.

### Input:
The adjacency Cost Matrix.

### Output:
MInimum cost path and Minimum Cost.

image (2)

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [x] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by shuvojitss**
### Description
This pull request is made to add a new algorithm shortest path in multistage graph. It calculates the shortest path between the source and the sink of the multistage graph and also gives the corresponding shortest distance value. It uses dynamic programming to calculate that. The program is w.r.t. issue no. #1279 which is raised to add the this program.

Fixes #1279

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shuvojitss** [comment]
This program is w.r.t issue #1279 which you assigned me.

File: docs/dynamic-programming/Multistage-graph.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+---
+id: MULTISTAGE GRAPH
```
Comment (ajay-dhangar) [inline docs/dynamic-programming/Multistage-graph.md]: Replace `id: MULTISTAGE GRAPH` with `id: multistage-graph`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Please read our [contribution guideline](https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md ) for a better understanding of our format and expectations. When writing content, please follow the guidelines to ensure that your contributions are accepted.

**Comment by shuvojitss** [comment]
@ajay-dhangar I have changed the id value as per required. If any other changes is required I will also do that. Please review it

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**